gql/builder.go

```diff
@@ -325,6 +342,58 @@ func (ob *ObjectBuilder) resolveObjectByName(p graphql.ResolveTypeParams) *graph
 	return ob.objects[name]
 }
 
+// ResolveListField returns a FieldResolveFn that leverages ResolveByField to get the field value then applies an
+// optional filter to the returned list of results.
+//
+// The optional filter is specified by a 'filter' argument which is a JSON object with a required string field
+// named 'Operation' and optional object field 'Argument' and string field named 'Field'. Certain operations may
+// require a valid 'Field' and/or 'Argument'. When provided 'Argument' should be a JSON object whose value will be
+// the argument to NewListOperation functions.
+//
+// Example:
+//  {
+//    query(id: "blah") {
+//      modules(filter: {Field: "moduleName", Operation: "==", Argument: {Value: "foo"}}) {
+//        moduleName
+//      }
+//    }
+//  }
+//
+func ResolveListField(name string, parent string) graphql.FieldResolveFn {
+	return func(p graphql.ResolveParams) (interface{}, error) {
+		filter, err := newListFilter(p.Args[filterArgumentName])
+		if err != nil {
+			return nil, err
+		}
+
+		resolve := ResolveByField(name, parent)
+
+		fullList, err := resolve(p)
```

Comment:
nit: we don't know that this is a list yet, so can we not call this fullList?

Reply:
fixing